samples/csharp/common/ObjectPool.cs

```diff
@@ -8,29 +8,44 @@ namespace Common
 {
     public class ObjectPool<T>
```

Comment:
@CESARDELATORRE - why are we creating our own ObjectPool class here? Why not use https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.objectpool.objectpool-1 instead?